File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        int sum = 0;
        int ans = -1e9;
        for(int it:nums){
            sum += it;
            ans = max(ans,sum);
            if(sum<0) sum = 0;
        }
        return ans;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        sum =0
        ans = -1e9
        for i in nums:
            sum += i
            ans = max(ans,sum)
            if sum<0:
                 sum=0
        return ans
        